pkg/cmd/pipeline/start.go

```diff
@@ -92,44 +109,210 @@ tkn pipeline start foo --param NAME=VALUE --resource source=scaffold-git  -s Ser
+func (opt *startOptions) run(pName string) error {
+	if !opt.Last {
+		if err := opt.getInput(pName); err != nil {
+			return err
+		}
+	}
+
+	return opt.startPipeline(pName)
+}
+
+func (opt *startOptions) getInput(pname string) error {
+	cs, err := opt.params.Clients()
+	if err != nil {
+		return err
+	}
+	pres, err := getPipelineResources(cs.Tekton, opt.params.Namespace())
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to list pipelineresources from %s namespace \n", opt.params.Namespace())
+		return err
+	}
+
+	resources := getPipelineResourcesByFormat(pres.Items)
+
+	pipeline, err := getPipeline(cs.Tekton, opt.params.Namespace(), pname)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to get pipeline %s from %s namespace \n", pname, opt.params.Namespace())
+		return err
+	}
+
+	err = opt.getInputResources(resources, pipeline)
+	if err != nil {
+		return err
+	}
+
+	err = opt.getInputParams(pipeline)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (opt *startOptions) getInputResources(resources []resourceOptions, pipeline *v1alpha1.Pipeline) error {
+	var ans string
+	for _, res := range pipeline.Spec.Resources {
+		options := getOptionsByFilter(resources, string(res.Type))
+		if len(options) == 0 {
+			fmt.Fprintln(opt.stream.Err,
+				fmt.Sprintf("No pipeline resource of type %s found in namespace: %s",
+					string(res.Type), opt.params.Namespace()))
+			return nil
+		}
+
+		var qs = []*survey.Question{
+			{
+				Name: "pipelineresource",
+				Prompt: &survey.Select{
+					Message: fmt.Sprintf("Which %s resource to use for %s ?", res.Type, res.Name),
+					Options: options,
+				},
+			},
+		}
+
+		if err := survey.Ask(qs, &ans, opt.askOpts); err != nil {
+			fmt.Println(err.Error())
+			return err
+		}
+
+		name := strings.TrimSpace(strings.Split(ans, ":")[0])
+		opt.Resources = append(opt.Resources, res.Name+"="+name)
+	}
+	return nil
+}
+
+func (opt *startOptions) getInputParams(pipeline *v1alpha1.Pipeline) error {
+	var ans string
+	for _, param := range pipeline.Spec.Params {
+
+		var qs = []*survey.Question{
+			{
+				Name: "pipeline param",
+				Prompt: &survey.Input{
+					Message: fmt.Sprintf("Value of  params %s ? (Default is %s)", param.Name, param.Default),
+				},
+			},
+		}
+
+		if err := survey.Ask(qs, &ans, opt.askOpts); err != nil {
+			fmt.Println(err.Error())
+			return err
+		}
+
+		opt.Params = append(opt.Params, param.Name+"="+ans)
+	}
+	return nil
+}
+
+func getPipelineResources(client versioned.Interface, namespace string) (*v1alpha1.PipelineResourceList, error) {
+	prec := client.TektonV1alpha1().PipelineResources(namespace)
+	pres, err := prec.List(v1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return pres, nil
+}
+
+func getPipelineResourcesByFormat(resources []v1alpha1.PipelineResource) (ret []resourceOptions) {
+	for _, res := range resources {
+		output := ""
+		if string(res.Spec.Type) == "git" {
```

Comment:
better to use a `switch` statement

Reply:
ok, cool I will do that way

Reply:
I have addressed this and updated the PR